osbenchmark/benchmark.py

```diff
@@ -863,6 +874,30 @@ def prepare_test_executions_dict(args, cfg):
                 test_executions_dict[execution] = None
     return test_executions_dict
 
+def run_and_aggregate(arg_parser, args, cfg):
+    semaphore = threading.Semaphore(1)
+    test_exes = []
+    args.subcommand = "execute-test"
+    aggregate_id = args.test_execution_id
+
+    for _ in range(args.test_iterations):
+        console.info(f"Running test {_ + 1}...")
+        args.test_execution_id = str(uuid.uuid4()) # we reuse the same args for each test so need to refresh the id
+        test_exes.append(args.test_execution_id)
+        with semaphore:
+            dispatch_sub_command(arg_parser, args, cfg)
+
+    console.info(f"Test executions: {', '.join(test_exes)}")
+    console.info("Aggregating results...")
+    aggregate_args = arg_parser.parse_args([
+    "aggregate",
+    f"--test-executions={','.join(test_exes)}",
+    f"--test-execution-id={aggregate_id}",
+    f"--results-file={args.results_file}",
+    f"--workload-repository={args.workload_repository}"
+    ])
+    dispatch_sub_command(arg_parser, aggregate_args, cfg)
```

Comment:
I think we should avoid using `dispatch_sub_command` again and parsing args here. Let's sync offline about this.